### ISSUE TITLE
add Math.choose 

### ISSUE THREAD

**Issue by adabson** [Onitz/npm-wetbox#1]
```
//this time its 45 Choose 6 = 45/6*44/5*43/4*42/3*41/2*40/1 = 8 145 060
function choose(n, k) {
  let ret = 1;
  
  if(k===0 || k===n) {
    return 1;
  } else if( k===1 ) {
    return n;
  } else if( k<0 || k>n) {
    return 0;
  } else if(k > n/2) {
    k = n - k;
  }

  for(let i=1;i<=k;i++){
    ret*=(n-k+i)/i;
  }
  return Math.round(ret);
}
```

**Comment by adabson** [comment]
elegant, simple

**Comment by adabson** [comment]
In addition to choose, add a combination generator function 
```
function kthCombination(k, arr, groupSize) {
  if( groupSize === 0 ) {
    return [];
  } else if( arr.length === groupSize ) {
    return arr;
  } else {
    let remainingSlots = choose(arr.length-1, groupSize-1);
    if( k < remainingSlots ) {
      return [arr[0]].concat( kthCombination(k, arr.slice(1), groupSize-1) );
    } 
    return kthCombination(k-remainingSlots, arr.slice(1), groupSize);
  }
}
```
https://stackoverflow.com/questions/1776442/nth-combination

![tree12345](https://user-images.githubusercontent.com/1431735/51154956-fdf9a400-18c9-11e9-9273-a386cb23f07a.png)

![simplified](https://user-images.githubusercontent.com/1431735/51158681-b7ac4100-18d9-11e9-876f-f1594231c022.png)

**Comment by adabson** [comment]
Finally, it'd also be amazeballs to have an inverse kth-combination generator
```
function inverseKthCombination(n, arr, k) { // k is the groupsize, ie (n choose k)
  //let n = widerArray.length; //should be able to switch out n with wider array evenutally
  arr.sort((a,b)=>a-b); // expect ascending order
  let widerArray = Array.from({length: n}, (x, y) => y+1); //consecutiveN
  let position = Array.from(arr, x => widerArray.indexOf(x));
  let ret = 0;

  for(let i=0;i<position.length;i++) {
    let last = i===0 ? 0 : position[i-1]+1;
    add = choose(n-last, k-i) - choose(n-position[i], k-i);
    ret += add;
  }

  return ret;
}
```

Preferably automate the testing. For large combination sizes like N=35 K=7, we require a math.round() in our choose function (as the alternative to computing large factorials, javascript starts to have a problem with decimal math for big numbers)
```
let success = 0;
for(let i=0;i<choose(N,K);i++) {
  let combo = kthCombination(i, consecutive(N), K);
  let invCombo = inverseKthCombination(N, combo, K);
  let isGood = i===invCombo;
  
  if(!isGood)console.log( i, combo, invCombo, isGood );
  if(isGood) {
    success++;
  }
}
console.log('----');
console.log('Test coverage:',((success/choose(N,K))*100).toFixed(2),`% passed (${success} passed, ${choose(N,K) - success} failed)`);
```

**Comment by adabson** [comment]
`choose(), ithCombination(), inverseIthCombination(), hasDuplicates()` are all included and live in npm-wetbox@0.3.5 !
